tensorflow_probability/examples/cifar10_bnn.py

```diff
@@ -0,0 +1,253 @@
+                         batch_size, valid_size):
+  """Build an Iterator switching between train and heldout data."""
+
+  x_train = x_train.astype('float32')
+  x_test = x_test.astype('float32')
+
+  x_train /= 255
+  x_test /= 255
+
+  y_train = y_train.flatten()
+  y_test = y_test.flatten()
+
+  print('x_train shape:' + str(x_train.shape))
+  print(str(x_train.shape[0]) + ' train samples')
+  print(str(x_test.shape[0]) + ' test samples')
+
+  # Build an iterator over training batches.
+  training_dataset = tf.data.Dataset.from_tensor_slices(
+      (x_train, np.int32(y_train)))
+  training_batches = training_dataset.shuffle(
+      50000, reshuffle_each_iteration=True).repeat().batch(batch_size)
+  training_iterator = training_batches.make_one_shot_iterator()
+
+  # Build a iterator over the heldout set with batch_size=heldout_size,
+  # i.e., return the entire heldout set as a constant.
+  heldout_dataset = tf.data.Dataset.from_tensor_slices(
+      (x_test, np.int32(y_test)))
+  heldout_batches = heldout_dataset.repeat().batch(valid_size)
+  heldout_iterator = heldout_batches.make_one_shot_iterator()
+
+  # Combine these into a feedable iterator that can switch between training
+  # and validation inputs.
+  handle = tf.placeholder(tf.string, shape=[])
+  feedable_iterator = tf.data.Iterator.from_string_handle(
+      handle, training_batches.output_types, training_batches.output_shapes)
+  images, labels = feedable_iterator.get_next()
+
+  return images, labels, handle, training_iterator, heldout_iterator
+
+
+def main(argv):
+  del argv  # unused
+  if tf.gfile.Exists(FLAGS.model_dir):
+    tf.logging.warning(
+        "Warning: deleting old log directory at {}".format(
+            FLAGS.model_dir))
+    tf.gfile.DeleteRecursively(FLAGS.model_dir)
+  tf.gfile.MakeDirs(FLAGS.model_dir)
+
+  (x_train, y_train), (x_test, y_test) = cifar10.load_data()
+
+  (images, labels, handle,
+   training_iterator,
+   heldout_iterator) = build_input_pipeline(x_train, x_test, y_train, y_test,
+                                            FLAGS.batch_size, 500)
+
+  if FLAGS.architecture == "resnet":
+    model_fn = bayesian_resnet
+  else:
+    model_fn = bayesian_vgg
+  
+  model = model_fn(IMAGE_SHAPE,
+                   num_classes=10,
+                   kernel_posterior_scale_mean=FLAGS.kernel_scale_mean,
+                   kernel_posterior_scale_constraint=FLAGS.kernel_scale_constraint)
+  logits = model(images)
+  labels_distribution = tfd.Categorical(logits=logits)
+
+  # Perform KL annealing. The optimal number of annealing steps 
+  # depends on the dataset and architecture.   
+  t = tf.Variable(0.0)
+  kl_regularizer = t / (FLAGS.kl_annealing * len(x_train) / FLAGS.batch_size)
+
+  # Compute the -ELBO as the loss, averaged over the batch size.
+  log_likelihood = labels_distribution.log_prob(labels)
+  neg_log_likelihood = -tf.reduce_mean(log_likelihood)
+  kl = sum(model.losses) / len(x_train) * tf.minimum(1.0, kl_regularizer)
```

Comment:
The proper ELBO doesn't scale by batch size on the KL, right? If I understand correctly, you should write a comment saying that you're scaling the KL term.